service/history/ndc/activity_state_replicator_test.go

```diff
@@ -93,6 +94,9 @@ func (s *activityReplicatorStateSuite) TearDownSuite() {
 func (s *activityReplicatorStateSuite) SetupTest() {
 	s.Assertions = require.New(s.T())
 
+	config := tests.NewDynamicConfig()
+	config.ShardFinalizerTimeout = dynamicconfig.GetDurationPropertyFn(0) // disabling Shard cleanup to keep tests simple
```

Comment:
Curious what's special about this test suite that requires cleanup logic to be disabled? Shall we disable it by default and only enable for tests that are specifically testing the clean up?

Reply:
Right; so this test suite (and the other one as well) will invoke `s.mockShard.StopForTest()` which eventually triggers the Finalizer to run. Which in and of itself is absolutely no problem. However, these test suites use a `workflow.MockContext` with a couple of `EXPECT`s. I just don't see any value in mocking the Lock, Unlock and Clear calls for every single test case. So I disabled them explicitly for these 2 tests suites.

Now, having said all this ... I think the better approach is to just remove the Finalizer from the mock shard entirely. So I'll do that next.